src/main/java/Parser.java

```diff
@@ -0,0 +1,67 @@
+public class Parser {
+    private String[] information;
+    private String date;
+    private Command command;
+    private String description;
+    private int taskIndex;
+
+    public Parser(String input) {
+        try {
+            this.information = input.split(" ", 2);
+            this.command = Command.valueOf(this.information[0].toUpperCase());
+
+            switch (this.command) {
+                case FIND:
+                    this.description = this.information[1];
+                    break;
+                case DELETE:
+                case DONE:
+                    this.taskIndex = Integer.parseInt(this.information[1]);
+                    break;
+                case TODO:
+                case DEADLINE:
+                case EVENT:
+                    String[] descriptionAndTime = information.length < 2 ? null : information[1].split("/", 2);
+                    this.description = descriptionAndTime == null ? "DESCRIPTION.ERROR" : descriptionAndTime[0];
+                    this.date = descriptionAndTime == null || descriptionAndTime.length < 2 ? "DATE.ERROR"
+                            : descriptionAndTime[1].split(" ", 2)[1];
```

Comment:
I like that this ternary was broken into two lines to keep within the 120 char limit but should it be formatted as 

(CONDITION)
? method1()
: method2()

instead?